src/Altinn.Common/Altinn.Common.AccessToken/Altinn.Common.AccessToken/AccessTokenHandler.cs

```diff
@@ -0,0 +1,145 @@
+        /// Handles verification of AccessTokens. Enabled with Policy on API controllers 
+        /// </summary>
+        /// <param name="context">The context</param>
+        /// <param name="requirement">The requirement for the given operation</param>
+        /// <returns></returns>
+        protected override async Task HandleRequirementAsync(AuthorizationHandlerContext context, AccessTokenRequirement requirement)
+        {
+            StringValues tokens = GetAccessTokens();
+            if (tokens.Count != 1 && _accessTokenSettings.DisableAccesTokenVerification)
+            {
+                context.Succeed(requirement);
+                await Task.CompletedTask;
+                return;
+            }
+
+            // It should only be one accesss token
+            if (tokens.Count != 1)
+            {
+                context.Fail();
+            }
+
+            bool isValid = false;
+            try
+            {
+                isValid = await ValidateAccessToken(tokens[0]);
+            }
+            catch(Exception ex)
+            {
+                _logger.LogWarning(ex, "Validation of Access Token Failed");
+                if (!_accessTokenSettings.DisableAccesTokenVerification)
+                {
+                    context.Fail();
+                }
+                else
+                {
+                    context.Succeed(requirement);
+                    await Task.CompletedTask;
+                    return;
+                }
+            }
+
+            if (isValid)
+            {
+                context.Succeed(requirement);
+            }
+            else
+            {
+                context.Fail();
+            }
+        }
+
+        /// <summary>
+        /// This validates the access token available in 
+        /// </summary>
+        /// <param name="token"></param>
+        /// <returns></returns>
+        private async Task<bool> ValidateAccessToken(string token)
+        {
+            JwtSecurityTokenHandler validator = new JwtSecurityTokenHandler();
+
+            if (!validator.CanReadToken(token))
+            {
+                return false;               
+            }
+
+            // Read JWT token to extract Issuer
+            JwtSecurityToken jwt = validator.ReadJwtToken(token);
+            TokenValidationParameters validationParameters = await GetTokenValidationParameters(jwt.Issuer);
+
+            ClaimsPrincipal principal;
+            SecurityToken validatedToken;
+            try
+            {
+                principal = validator.ValidateToken(token, validationParameters, out validatedToken);
+                return true;
+            }
+            catch (Exception ex)
+            {
+                _logger.LogInformation("Failed to validate token.", ex);
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Gets 
+        /// </summary>
+        /// <param name="issuer"></param>
+        /// <returns></returns>
```

Comment:
Comments with no value. I often simply choose not to have a XmlComments on private methods if I think the method is clear without.

Reply:
done